content/articles/eclat-algorithm-in-python/index.md

```diff
@@ -0,0 +1,154 @@
+
+1. Apriori Algorithm
+2. Eclat Algorithm
+3. FP Growth Algorithm
+
+The first algorithm to be introduced in the Data Mining domain was the Apriori algorithm. This algorithm, however, had some limitations in discovering frequent itemsets. Its limitations created a need for a more efficient algorithm.
+
+Later, the Eclat algorithm was introduced to deal with the weakness of the Apriori algorithm. Between these two algorithms, we have a significant difference.
+
+The first difference is that, unlike the Apriori algorithm, which is applicable with *horizontal dataset*, the Eclat algorithm is applicable only with a dataset in *verticle dataset* format.
+
+Secondly, in the Eclat algorithm, only the *support* is counted, and the confidence, as in the case of Apriori, is not computed. Here, the *Support* is nothing but the number of times an item is in a database.
+
+At each stage of the generated database, the Eclat algorithm uses the current generated dataset to learn frequent itemset, unlike the Apriori, which scans the original database repeatedly. Since the Eclat scan over the database once, it is much faster than the Apriori algorithm.
+
+However, this doesn't mean the Apriori algorithm is worse. On the contrary, when dealing with a larger dataset, Apriori tends to shine best. Therefore, the Eclat algorithm works better with small and medium datasets.
+
+The key takeaway here is that Eclat works with a vertical data format. Since most datasets are in horizontal format, to apply the Eclat algorithm, we first have to convert them to vertical format.
+
+Below are examples of Horizontal and verticle data formats;
+
+![Horizontal data](/engineering-education/eclat-algorithm-in-python/horizontal.png)
+Horizontal data format
+
+![Verticle data](/engineering-education/eclat-algorithm-in-python/verticle-format.png)
+Verticle data format
+
+We obtain a verticle data format by making a list of transactions to which each particular item is found.
+
+Let's look at the steps in the Eclat algorithm.
+
+### Eclat Algorithm
+1. Get tidlist for each item.
+ Here we need to scan the entire database. The tidlist of item {a} is the list of all transactions in which item {a} is found.
+2. Intersect tidlist of item {a} with tidlist of item {b} and generate a new tansaction list that both items {a} and {b} are found.
+3. Reapet step 1 on {a}-coditional database.
+4. For all other items, repeat the above steps.
+
+If we perform these steps on the verticle dataset we gave above, we should obtain an output similar to the one below;
+![data image](/engineering-education/eclat-algorithm-in-python/tid-set.png)
+
+
+### Advantages
+1. Eclat algorithm has low memory requirements compared to Apriori as it uses a Depth-First Search approach.
```

Comment:
You can either explain what "Depth-First Search" approach is or add an external link to it.